File: pms-rest-api/src/main/kotlin/com/seg3502/pms/entities/Prescription.kt
package com.seg3502.pms.entities

import javax.persistence.*
import javax.validation.constraints.Min
import javax.validation.constraints.NotEmpty

@Entity
class Prescription() {
    constructor(patient: Patient, medicationNumber: String, medicationName: String) : this() {
        this.patient = patient
        this.medicationNumber = medicationNumber
        this.medicationName = medicationName
    }

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0

    @ManyToOne
    var patient: Patient = Patient()

    @NotEmpty
    var medicationNumber: String = ""

    @NotEmpty
    var medicationName: String = ""

    @OneToMany(cascade = [CascadeType.ALL], fetch = FetchType.LAZY, mappedBy = "prescription")
    var schedule: MutableList<PrescriptionSchedule> = ArrayList()
}

File: pms-frontend/src/app/prescriptions/model/prescription.ts
import { Patient } from '../../patients/model/patient';
import { PrescriptionSchedule } from './prescription-schedule';

export class Prescription {
  id: number;
  patient: Patient;
  medicationNumber: string;
  medicationName: string;
  unitsPerDay: number;
  administrationsPerDay: number;
  schedule: PrescriptionSchedule[];
}
